{"cells": [{"metadata": {}, "id": "9f902f7f", "cell_type": "code", "source": "from tensorflow.keras.preprocessing.image import ImageDataGenerator\nimport tensorflow as tf\nimport matplotlib.pyplot as plt\nfrom tensorflow.keras import layers,models\nimport os\nimport numpy as np", "execution_count": 1, "outputs": []}, {"metadata": {}, "id": "5c0eafb6", "cell_type": "code", "source": "pwd", "execution_count": 3, "outputs": [{"output_type": "execute_result", "execution_count": 3, "data": {"text/plain": "'/home/wsuser/work'"}, "metadata": {}}]}, {"metadata": {}, "id": "279e9e15", "cell_type": "code", "source": "import os, types\nimport pandas as pd\nfrom botocore.client import Config\nimport ibm_boto3\n\ndef __iter__(self): return 0\n\n# @hidden_cell\n# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.\n# You might want to remove those credentials before you share the notebook.\ncos_client = ibm_boto3.client(service_name='s3',\n    ibm_api_key_id='oMGmiKgTzljtiNY3Mun5dF2KggJ_qxkEHJRiiCYrOec8',\n    ibm_auth_endpoint=\"https://iam.cloud.ibm.com/oidc/token\",\n    config=Config(signature_version='oauth'),\n    endpoint_url='https://s3.private.us.cloud-object-storage.appdomain.cloud')\n\nbucket = 'imageclassification-donotdelete-pr-tljphflac3cpyb'\nobject_key = 'Veg-dataset.zip'\n\nstreaming_body_2 = cos_client.get_object(Bucket=bucket, Key=object_key)['Body']\n\n# Your data file was loaded into a botocore.response.StreamingBody object.\n# Please read the documentation of ibm_boto3 and pandas to learn more about the possibilities to load the data.\n# ibm_boto3 documentation: https://ibm.github.io/ibm-cos-sdk-python/\n# pandas documentation: http://pandas.pydata.org/\n", "execution_count": 4, "outputs": []}, {"metadata": {}, "id": "80299455", "cell_type": "code", "source": "from io import BytesIO\nimport zipfile\nunzip = zipfile.ZipFile(BytesIO(streaming_body_2.read()), 'r')\nfile_paths = unzip.namelist()\nfor path in file_paths:\n    unzip.extract(path)", "execution_count": 5, "outputs": []}, {"metadata": {}, "id": "c42325d9", "cell_type": "code", "source": "pwd", "execution_count": 10, "outputs": [{"output_type": "execute_result", "execution_count": 10, "data": {"text/plain": "'/home/wsuser/work'"}, "metadata": {}}]}, {"metadata": {}, "id": "e3b2b786", "cell_type": "code", "source": "train_datagen = ImageDataGenerator (rescale = 1./255, shear_range= 0.2,zoom_range= 0.2, horizontal_flip = True)\ntest_datagen =ImageDataGenerator (rescale = 1)", "execution_count": 6, "outputs": []}, {"metadata": {}, "id": "14d42bbe", "cell_type": "code", "source": "x_train = train_datagen.flow_from_directory(\n        r'D:\\IBM\\Dataset Plant Disease\\fruit-dataset\\fruit-dataset\\train',\n        target_size=(IMAGE_SIZE,IMAGE_SIZE),\n        batch_size=32,\n        class_mode=\"sparse\",\n#         save_to_dir=\"C:\\\\Code\\\\potato-disease-classification\\\\training\\\\AugmentedImages\"\n)", "execution_count": 15, "outputs": [{"name": "stdout", "output_type": "stream", "text": "Found 5384 images belonging to 6 classes.\n"}]}, {"metadata": {}, "id": "62d74bf6", "cell_type": "code", "source": "x_train = train_datagen.flow_from_directory(r'/home/wsuser/work/Veg-dataset/train_set',target_size = (128,128), batch_size = 32, class_mode = 'categorical')\nx_test = test_datagen.flow_from_directory(r'/home/wsuser/work/Veg-dataset/test_set',target_size = (128,128), batch_size = 32, class_mode = 'categorical')", "execution_count": 7, "outputs": [{"output_type": "stream", "text": "Found 11386 images belonging to 9 classes.\nFound 3416 images belonging to 9 classes.\n", "name": "stdout"}]}, {"metadata": {}, "id": "c6e732c7", "cell_type": "code", "source": "model.save('fruit')", "execution_count": 14, "outputs": [{"name": "stdout", "output_type": "stream", "text": "INFO:tensorflow:Assets written to: fruit\\assets\n"}]}, {"metadata": {}, "id": "d576b44a", "cell_type": "code", "source": "pip install 'h5py==2.10.0' --force-reinstall", "execution_count": 10, "outputs": [{"name": "stdout", "output_type": "stream", "text": "Note: you may need to restart the kernel to use updated packages.\n"}, {"name": "stderr", "output_type": "stream", "text": "ERROR: Invalid requirement: \"'h5py==2.10.0'\"\n"}]}, {"metadata": {}, "id": "fb716e72", "cell_type": "code", "source": "x_train=train_datagen.flow_from_directory(r\"D:\\IBM\\Dataset Plant Disease\\fruit-dataset\\fruit-dataset\\train\",target_size=(128,128),\n                                        class_mode='categorical',batch_size=24)", "execution_count": 47, "outputs": [{"name": "stdout", "output_type": "stream", "text": "Found 5384 images belonging to 6 classes.\n"}]}, {"metadata": {}, "id": "8b8d8485", "cell_type": "code", "source": "\n\nfrom tensorflow.keras.models import Sequential\nfrom tensorflow.keras.layers import Dense,Convolution2D,MaxPooling2D,Flatten\n\n", "execution_count": 8, "outputs": []}, {"metadata": {}, "id": "e6b5f2af", "cell_type": "code", "source": "", "execution_count": null, "outputs": []}, {"metadata": {}, "id": "59dd4524", "cell_type": "code", "source": "\n\nmodel=Sequential()\n\n", "execution_count": 9, "outputs": []}, {"metadata": {}, "id": "ed230326", "cell_type": "code", "source": "model.add(Convolution2D(32,(3,3),input_shape=(128,128,3),activation='relu'))", "execution_count": 10, "outputs": []}, {"metadata": {}, "id": "6aaa20c6", "cell_type": "code", "source": "model.add(MaxPooling2D(pool_size=(2,2)))\nmodel.add(Flatten())\nmodel.summary()", "execution_count": 11, "outputs": [{"output_type": "stream", "text": "Model: \"sequential\"\n_________________________________________________________________\n Layer (type)                Output Shape              Param #   \n=================================================================\n conv2d (Conv2D)             (None, 126, 126, 32)      896       \n                                                                 \n max_pooling2d (MaxPooling2D  (None, 63, 63, 32)       0         \n )                                                               \n                                                                 \n flatten (Flatten)           (None, 127008)            0         \n                                                                 \n=================================================================\nTotal params: 896\nTrainable params: 896\nNon-trainable params: 0\n_________________________________________________________________\n", "name": "stdout"}]}, {"metadata": {}, "id": "679b3150", "cell_type": "code", "source": "model.add(Dense(300,activation='relu'))\nmodel.add(Dense(150,activation='relu'))", "execution_count": 12, "outputs": []}, {"metadata": {}, "id": "9261cd63", "cell_type": "code", "source": "model.add(Dense(9,activation='softmax'))\nmodel.compile(loss='categorical_crossentropy',optimizer='adam',metrics=['accuracy'])", "execution_count": 13, "outputs": []}, {"metadata": {}, "id": "ab268091", "cell_type": "code", "source": "len(x_train)", "execution_count": 14, "outputs": [{"output_type": "execute_result", "execution_count": 14, "data": {"text/plain": "356"}, "metadata": {}}]}, {"metadata": {}, "id": "f191ada6", "cell_type": "code", "source": "model.fit(x_train,steps_per_epoch=len(x_train),epochs=10)", "execution_count": 15, "outputs": [{"output_type": "stream", "text": "Epoch 1/10\n356/356 [==============================] - 106s 295ms/step - loss: 1.4770 - accuracy: 0.5919\nEpoch 2/10\n356/356 [==============================] - 105s 295ms/step - loss: 0.6183 - accuracy: 0.7857\nEpoch 3/10\n356/356 [==============================] - 105s 295ms/step - loss: 0.4647 - accuracy: 0.8358\nEpoch 4/10\n356/356 [==============================] - 105s 293ms/step - loss: 0.3753 - accuracy: 0.8700\nEpoch 5/10\n356/356 [==============================] - 105s 294ms/step - loss: 0.3028 - accuracy: 0.8978\nEpoch 6/10\n356/356 [==============================] - 105s 294ms/step - loss: 0.2970 - accuracy: 0.8944\nEpoch 7/10\n356/356 [==============================] - 105s 294ms/step - loss: 0.2455 - accuracy: 0.9130\nEpoch 8/10\n356/356 [==============================] - 105s 295ms/step - loss: 0.2358 - accuracy: 0.9187\nEpoch 9/10\n356/356 [==============================] - 106s 297ms/step - loss: 0.2190 - accuracy: 0.9232\nEpoch 10/10\n356/356 [==============================] - 105s 295ms/step - loss: 0.1793 - accuracy: 0.9389\n", "name": "stdout"}, {"output_type": "execute_result", "execution_count": 15, "data": {"text/plain": "<keras.callbacks.History at 0x7fb1920e0250>"}, "metadata": {}}]}, {"metadata": {}, "id": "8147bd7c", "cell_type": "code", "source": "model.save('VegModelWatson.h5')", "execution_count": 17, "outputs": []}, {"metadata": {}, "id": "db93f188", "cell_type": "code", "source": "!tar -zcvf IBM_TrainedModel1.tgz VegModelWatson.h5", "execution_count": 18, "outputs": [{"output_type": "stream", "text": "VegModelWatson.h5\r\n", "name": "stdout"}]}, {"metadata": {}, "id": "91c82e20", "cell_type": "code", "source": "!ls", "execution_count": 24, "outputs": [{"output_type": "stream", "text": "fruit-dataset  fruitModelWatson  IBM_TrainedModel.tgz\r\n", "name": "stdout"}]}, {"metadata": {}, "id": "eb8fb721", "cell_type": "code", "source": "!pip install watson-machine-learning-client --upgrade", "execution_count": 27, "outputs": [{"output_type": "stream", "text": "Collecting watson-machine-learning-client\n  Downloading watson_machine_learning_client-1.0.391-py3-none-any.whl (538 kB)\n\u001b[K     |\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588\u2588| 538 kB 22.0 MB/s eta 0:00:01\n\u001b[?25hRequirement already satisfied: lomond in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (0.3.3)\nRequirement already satisfied: tabulate in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (0.8.9)\nRequirement already satisfied: tqdm in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (4.62.3)\nRequirement already satisfied: ibm-cos-sdk in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (2.11.0)\nRequirement already satisfied: requests in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (2.26.0)\nRequirement already satisfied: urllib3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (1.26.7)\nRequirement already satisfied: boto3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (1.18.21)\nRequirement already satisfied: pandas in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (1.3.4)\nRequirement already satisfied: certifi in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from watson-machine-learning-client) (2022.9.24)\nRequirement already satisfied: botocore<1.22.0,>=1.21.21 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from boto3->watson-machine-learning-client) (1.21.41)\nRequirement already satisfied: s3transfer<0.6.0,>=0.5.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from boto3->watson-machine-learning-client) (0.5.0)\nRequirement already satisfied: jmespath<1.0.0,>=0.7.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from boto3->watson-machine-learning-client) (0.10.0)\nRequirement already satisfied: python-dateutil<3.0.0,>=2.1 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from botocore<1.22.0,>=1.21.21->boto3->watson-machine-learning-client) (2.8.2)\nRequirement already satisfied: six>=1.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from python-dateutil<3.0.0,>=2.1->botocore<1.22.0,>=1.21.21->boto3->watson-machine-learning-client) (1.15.0)\nRequirement already satisfied: ibm-cos-sdk-core==2.11.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from ibm-cos-sdk->watson-machine-learning-client) (2.11.0)\nRequirement already satisfied: ibm-cos-sdk-s3transfer==2.11.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from ibm-cos-sdk->watson-machine-learning-client) (2.11.0)\nRequirement already satisfied: charset-normalizer~=2.0.0 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests->watson-machine-learning-client) (2.0.4)\nRequirement already satisfied: idna<4,>=2.5 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from requests->watson-machine-learning-client) (3.3)\nRequirement already satisfied: pytz>=2017.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from pandas->watson-machine-learning-client) (2021.3)\nRequirement already satisfied: numpy>=1.17.3 in /opt/conda/envs/Python-3.9/lib/python3.9/site-packages (from pandas->watson-machine-learning-client) (1.20.3)\nInstalling collected packages: watson-machine-learning-client\nSuccessfully installed watson-machine-learning-client-1.0.391\n", "name": "stdout"}]}, {"metadata": {}, "id": "e9497bfb", "cell_type": "code", "source": "from ibm_watson_machine_learning import APIClient\nwml_credentials = {\n    \"url\": \"https://us-south.ml.cloud.ibm.com\",\n    \"apikey\": \"TAMGDjl2zVuiL_u957FqwDQSRSGpEediiqRVP6o_8zHS\"\n}\n\nclient = APIClient(wml_credentials)", "execution_count": 19, "outputs": []}, {"metadata": {}, "id": "39d6c25b", "cell_type": "code", "source": "import os, glob\nimport numpy as np\nimport seaborn as sns\nimport matplotlib.pyplot as plt\nimport pandas as pd\n\nfrom sklearn.model_selection import train_test_split\nfrom tensorflow.keras.preprocessing.image import ImageDataGenerator\nfrom tensorflow.keras.models import Model\nfrom tensorflow.keras.layers import Conv2D, MaxPooling2D, Flatten, Dense, Dropout\nfrom tensorflow.keras.callbacks import Callback,EarlyStopping\nfrom tensorflow.keras.applications import ResNet50\nfrom tensorflow.keras.applications.resnet50 import preprocess_input\nfrom sklearn.metrics import  classification_report", "execution_count": 63, "outputs": []}, {"metadata": {}, "id": "8cb4005f", "cell_type": "code", "source": "IMG_SHAPE = (224, 224)\nINPUT_SHAPE = [224, 224, 3]\nEPOCHS = 10\nBS = 32", "execution_count": 66, "outputs": []}, {"metadata": {}, "id": "9ad9a29d", "cell_type": "code", "source": "from tensorflow.keras.applications.vgg16 import VGG16", "execution_count": 71, "outputs": []}, {"metadata": {}, "id": "56218421", "cell_type": "code", "source": "pip install h5py", "execution_count": 74, "outputs": [{"name": "stdout", "output_type": "stream", "text": "Requirement already satisfied: h5py in c:\\users\\abiz1\\.conda\\envs\\tf_test\\lib\\site-packages (3.7.0)\nRequirement already satisfied: numpy>=1.14.5 in c:\\users\\abiz1\\.conda\\envs\\tf_test\\lib\\site-packages (from h5py) (1.23.4)\nNote: you may need to restart the kernel to use updated packages.\n"}]}, {"metadata": {}, "id": "efc4d2a7", "cell_type": "code", "source": "", "execution_count": null, "outputs": []}], "metadata": {"kernelspec": {"name": "python3", "display_name": "Python 3.9", "language": "python"}, "language_info": {"name": "python", "version": "3.9.13", "mimetype": "text/x-python", "codemirror_mode": {"name": "ipython", "version": 3}, "pygments_lexer": "ipython3", "nbconvert_exporter": "python", "file_extension": ".py"}}, "nbformat": 4, "nbformat_minor": 5}